### PR TITLE
Support non json formats + file encryption

### PR DESCRIPTION
**Pull request by shreyasthiagaraj**
This PR adds support for non JSON formats in getFile and putFile, specifically "text/plain" and "application/octet-stream" content types. It also adds support for encryption and decryption of data. Finally, it solves a bug that was surfaced while testing this feature where files cannot be shared between a web app and corresponding iOS app due to mismatched Gaia addresses.

It will solve issues #12, #16, #21, #39, and #44.

### CODE REVIEW

**Review by yknl — COMMENTED** [review]
Great work! It works really well when I tried it out. Would love to see test coverage for these changes.

**Comment by shreyasthiagaraj** [comment]
@yknl Submitted tests 👍🏾.

- Gaia__without_encryption__for_text_content__can_upload_and_retrieve
- Gaia__without_encryption__for_bytes_content__can_upload_and_retrieve
- Gaia__with_encryption__can_upload_and_retrieve
- Gaia__with_encryption__fails_retrieve_without_decrypt

They cover all content types (json, plain text, bytes) and encryption/no encryption cases.

**Review by larrysalibra — CHANGES_REQUESTED** [review]
@shreyasthiagaraj from what I can tell of the tests this seems to work. 

It would be nice if the example application was updated to demonstrate the additional functionality similar to what we have in Android.

I'd also like to see documentation of the form of api comments. For example, it's not immediately apparent from looking at the `getFile` signature how exactly one would use completion handler.

**Comment by shreyasthiagaraj** [comment]
@larrysalibra done! Comments + sample app update. The sample app still needs more work to look like Android's, but that is tracked elsewhere (#46 ) and is outside the scope of this PR.

File: Blockstack/Classes/Blockstack.swift [hunk 1]
```diff
@@ -101,7 +101,16 @@ open class Blockstack {
         self.sfAuthSession?.start()
     }
     
-    public func putFile(to path: String, content: String, encrypt: Bool = false, completion: @escaping (String?, GaiaError?) -> Void) {
+    /**
```
Comment (larrysalibra) [inline Blockstack/Classes/Blockstack.swift]: Love these! Beautiful!

**Review by larrysalibra — APPROVED** [review]
This works well for me! Awesome job @shreyasthiagaraj !

**Comment by larrysalibra** [comment]
I tested this out by modifying the example to store bytes instead of a string and make sure that I could read it back.

**Pull request merged**